src/shapes/Circle.ts

```diff
@@ -38,6 +45,10 @@ export class Circle extends FabricObject {
     'endAngle',
   ];
 
+  constructor(options?: CircleProps) {
+    super(options);
+  }
```

Comment:
If i m not mistaken as long as the  interface is fulfilled, we can add any property right?
foo, bar, baz are still valid. correct?

Reply:
good question
the answer is no


```ts
interface A {
  foo: 'bar'
}

function actStrict(a:A){

}

function act(a:A & Record<string,unknown>){

}

actStrict({ foo:'bar', baz: 'pi' })
act({ foo:'bar', baz: 'pi' })
```



<details><summary><b>Output</b></summary>

```ts
"use strict";
function actStrict(a) {
}
function act(a) {
}
actStrict({ foo: 'bar', baz: 'pi' });
act({ foo: 'bar', baz: 'pi' });

```


</details>


<details><summary><b>Compiler Options</b></summary>

```json
{
  "compilerOptions": {
    "strict": true,
    "noImplicitAny": true,
    "strictNullChecks": true,
    "strictFunctionTypes": true,
    "strictPropertyInitialization": true,
    "strictBindCallApply": true,
    "noImplicitThis": true,
    "noImplicitReturns": true,
    "alwaysStrict": true,
    "esModuleInterop": true,
    "declaration": true,
    "experimentalDecorators": true,
    "emitDecoratorMetadata": true,
    "target": "ES2017",
    "jsx": "react",
    "module": "ESNext",
    "moduleResolution": "node"
  }
}
```


</details>

**Playground Link:** [Provided](https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgILIN4FgBQzkwD2hAXMgOQBGcU5uAvrrjAK4gJjCEjKJgDKYKMA4AKOCVQBKbDgZMcrdp268xE9ADJkAJQgJCUACYAeAM5DQAcwA0bANYhCAdxAA+GQsZycfQcLEMAmISKhpyG2RqAC8ycgAHYHJkeilcPlEgolIw2kiYuMTk1KA)

Reply:
I was about to merge this, but this blocks me.
I want to be able to do `Rect({ name: 'myname', id: '33' })` i think is a basic right was here before we introduced TS and shouldn't go away.

Reply:
Give me a few mins to take a look